### DISCUSSION TITLE
Temporary Overrides not working in Nightscout with Google Cloud

### DISCUSSION THREAD

**Discussion by Jsoto250** [NightscoutFoundation/xDrip#2590] [Q&A]
Hello, 
I have followed all directions and added the necessary config vars but for some reason the temporary override doesn't show up in my nightscout. I was working on this about 2 nights ago and left it alone as I saw in other posts that after a few hours it would show up, but no luck for me. I have started a temporary override on my loop app itself and then turned it off but still nothing. I will attach a picture of my config vars, maybe something wrong with the key? Running NightScout 14.2.6. I did run out of sensors and should be receiving them in the mail today because I forgot to call in a failed sensor and my 3 month supply ran out. Wondering if it could be because of a loop not completing successfully since I havre no active cgm sensor? TIA!
![Vars](https://user-images.githubusercontent.com/63935530/210112655-38a21b2a-631a-4031-9550-c7301f54690e.JPG)

**Comment by Navid200** [comment]
This is a key for supporting Apple devices, right?
There must be a group where other Apple users and developers can communicate.   Do you not have access to such a group?

Nightscout on Google Cloud runs Nightscout as is.  We have not modified Nightscout.  

The only difference with Nightscout you ran on Heroku is that the authentication setting default for Heroku has been left open to the world for reading.  We have changed that to deny access to the world.  So, even to read, you either need to use the API_SECRET or a token.  other than that nothing else is different.

This group is knowledgeable about Loop, which I believe uses iOS:  https://www.facebook.com/groups/LOOPandLEARN

**Reply by Jsoto250** [reply]
This is so that I can set temporary overrides on Nightscout itself from a browser. I do have Loop on an apple device. I have posted in the cgm in the cloud group and the loop group, hoping to get an answer. I saw a thread with this same issue and I believe you had helped with the key. I didn't want to hijack his post lol.

**Reply by Navid200** [reply]
Please provide a link to your question in the Loop and Learn group.
I provided a link to it above.

**Reply by Jsoto250** [reply]
I had placed it in the CGM in the cloud group and thought I placed it in "looped" group not the Loop and Learn. I had posted another question in there. I just posted in loop and learn.

https://www.facebook.com/groups/cgminthecloud/permalink/5803706229750585/

https://www.facebook.com/groups/LOOPandLEARN/permalink/3328585177397963/

**Reply by Navid200** [reply]
Have you [restarted Nightscout](https://navid200.github.io/xDrip/docs/Nightscout/Restart.html) after editing the variables?

**Reply by Jsoto250** [reply]
Yes, I rebooted the virtual machine. I wonder if the loop apns key needs to be all on one line? I copy and pasted directly from heroku because I saw that on another thread with the same issue.

**Reply by Navid200** [reply]
I believe it should be on multiple lines.

But, I wonder if the app that had generated the original key, that you used on Heroku, has updated the key and you need to get the key from that app instead of from Heroku.

**Reply by Jsoto250** [reply]
Ok thank you.

I actually deleted the current key and created a new one when I was setting up nightscout on google cloud. When I saw your post about it being on separate lines, originally the vars line was all one long line because I had copied into the var config web tool. So what I did was I copy and pasted the new one I created, into heroku, and then copy and pasted it directly into the vars config on the virtual machine using nano directly from heroku.

**Reply by Navid200** [reply]
Was this ever resolved?

**Reply by Jsoto250** [reply]
Unfortunately, no. I never heard back from anyone after they asked a few initial questions.

**Reply by Navid200** [reply]
@psonnera @tzachi-dar You guys have this working?  Can you help?

**Reply by psonnera** [reply]
@Navid200 not using this feature. Seen something similar in FB but can't find the post.

**Comment by tzachi-dar** [comment]
@Jsoto250 as far as I remember the private key should have 3 lines of actual text (you have 4 in your picture).
In any case, please logon to your google machine, and choose the update option.
After that you will have a web server that supports multi lines.
Please paste your key using the new web server and choose reboot server. Wait 3 more hours and let's see if this will work.

Please also make sure that on the nightscout settings (in chrome) you have selected everything that applies.

**Reply by Jsoto250** [reply]
Ah I will try this. Originally I had pasted it into the web tool but it pasted the key as one long string. I opted to do it through nano editor and not the web tool and that one pasted it as what you see now. 

Let me update now and I will wait 3 hours. Thank you, will keep you posted!

**Reply by Jsoto250** [reply]
So I copy and pasted the key after opening it in Notepad. But now my variable looks like one long line. Should I be doing this on a mac so it keeps the formatting? I thought I read on another post with the same issue that the web tool doesn't work well for the key.
![Vars](https://user-images.githubusercontent.com/63935530/211452714-64131387-c03b-4e16-91cb-dbd2e95cd409.JPG)

**Reply by Navid200** [reply]
Would you please post a screenshot of the status page to make sure you have updated to the latest release?

**Reply by Jsoto250** [reply]
Sure, Is this the page? 
![Status](https://user-images.githubusercontent.com/63935530/211453652-d6071850-17e5-4d1d-a18c-458b311efa5e.JPG)

**Reply by Jsoto250** [reply]
Or this one? 
![Status](https://user-images.githubusercontent.com/63935530/211453886-9747b117-7151-4105-bed5-d0ea7d0f4658.JPG)

**Reply by Navid200** [reply]
Yes.  But, I didn't change the version number!
Would you please bring up the variable editor in a browser using item number 5 on the main menu?

**Reply by Navid200** [reply]
Does it look like the third image on this page?
https://navid200.github.io/xDrip/docs/Nightscout/NS_Variables

**Reply by Jsoto250** [reply]
It does not. It looks like this. 
![2F78155A-11C2-4D0F-954C-BF2CD3B61752](https://user-images.githubusercontent.com/63935530/211454506-563cc958-2578-4ed6-9ef7-2f8a1cc92cc6.jpeg)

**Reply by Navid200** [reply]
Please choose item 7 from the main menu and update.  Then, try again.  Does it look the same then?

**Reply by Jsoto250** [reply]
Yes! Now it looks like pic number 3. I edited the variable and now it looks like this. 
![D93E2050-5D26-4AF0-9D01-A73767232848](https://user-images.githubusercontent.com/63935530/211455178-5efc37f5-ad5e-45fd-a2f9-a3f7d970c9e6.jpeg)
![CFC133D7-D95A-40A3-AEFD-1D965FE4B5C2](https://user-images.githubusercontent.com/63935530/211455186-1c0415b2-849e-43ae-962c-abebbe14dacf.jpeg)

**Reply by Navid200** [reply]
Why does it have 4 lines?  According to Tzachi, it should have only 3!

**Reply by Jsoto250** [reply]
I don't understand why. Maybe I should try it on my mac. I am on my windows machine right now opening the key with notepad.

**Reply by Navid200** [reply]
Well, this also shows 4 lines:  https://loopkit.github.io/loopdocs/nightscout/remote-overrides/
So, I'm not sure.

**Reply by Jsoto250** [reply]
Hmmmm, I can try and paste in heroku cofig var and then copy and paste into the web tool and see if that makes a difference.

**Reply by Navid200** [reply]
I don't know if this is relevant to you.  But, have you see on the page I linked info about:
LOOP_PUSH_SERVER_ENVIRONMENT

**Reply by Jsoto250** [reply]
So, I have never been able to successfully build loop with testflight. I get to the last step which is building and I get a bunch of errors. I dont have that variable set, but should I? The version of loop on my phone was built on my mac with xcode.

**Reply by Jsoto250** [reply]
I tried updating the variable on my mac but it is still the same amount of lines.

**Reply by Jsoto250** [reply]
@tzachi-dar I updated my nightscout last night and got the new web view for editing configs. I waited until this morning and still no options. 

Just now I deleted the key and recreated a new one in Apple developer and deleted the config vars and recreated with new key. Going to wait a few hours to see if it resolves. My key still is showing up as 4 rows though. 
![image](https://user-images.githubusercontent.com/63935530/211685046-e1c6df10-dc4d-4f55-bfc1-251478e4d753.jpg)

**Reply by Jsoto250** [reply]
@tzachi-dar I waited a few days but creating a new key didn't seem to resolve the issue. :(

**Comment by tzachi-dar** [comment]
@Jsoto250 two things, when you post your keys like that, you practically give everyone access to your system. I suggest you change your keys in any way.
Another thing, if this has worked before when you have been using heroku, let's try and copy the parameters from there:
If you have not used two factor authentication on heroku:
please go to the command line and try running the script:
sudo /xDrip/scripts/GetHerokuVars.sh
Then if it looks good, run 
sudo cp heroku_vars.txt /etc/nsconfig
sudo reboot
Wait 2 hours and let's see if this works.

**Reply by Jsoto250** [reply]
@tzachi-dar  Thank you, I will create a new key again. I had tried to get it working on Heroku a while ago but it never showed up either. 

So I should create a new key, paste it into Heroku, and then run the commands in the terminal and wait 2 hours?

**Reply by Jsoto250** [reply]
@tzachi-dar @Navid200 I think it’s working now! I deployed Loop 3.0 tonight with TestFlight and added the additional variable needed and now I see the remote options. Not sure if I’m missing any but thanks a lot for your help! 
![D3BC6277-49EA-4B6F-AEA1-8243D76283D9](https://user-images.githubusercontent.com/63935530/212823786-dabee96f-7fff-4a57-b0ba-e93ecc0adbb1.png)

**Reply by Navid200** [reply]
Thanks for letting us know.  I am so happy it finally worked out.

**Reply by Jsoto250** [reply]
@Navid200 Thanks so much!!
